src/nft/components/collection/Card.tsx

```diff
@@ -277,13 +297,46 @@ function getHeightFromAspectRatio(uniformAspectRatio: UniformAspectRatio, render
     : renderedHeight
 }
 
+function getMediaAspectRatio(
+  uniformAspectRatio?: UniformAspectRatio,
+  setUniformAspectRatio?: (uniformAspectRatio: UniformAspectRatio) => void
+): string {
+  return uniformAspectRatio === UniformAspectRatios.square || !setUniformAspectRatio ? '1' : 'auto'
+}
+
 interface ImageProps {
   uniformAspectRatio?: UniformAspectRatio
   setUniformAspectRatio?: (uniformAspectRatio: UniformAspectRatio) => void
   renderedHeight?: number
   setRenderedHeight?: (renderedHeight: number | undefined) => void
 }
 
+const StyledMediaContainer = styled(Row)`
+  overflow: hidden;
+  border-top-left-radius: ${BORDER_RADIUS}px;
+  border-top-right-radius: ${BORDER_RADIUS}px;
+`
+
+const StyledImage = styled.img<{
+  uniformAspectRatio?: UniformAspectRatio
+  setUniformAspectRatio?: (uniformAspectRatio: UniformAspectRatio) => void
+  renderedHeight?: number
+  hovered: boolean
+  imageLoading: boolean
+  $hidden?: boolean
+}>`
+  width: 100%;
+  aspect-ratio: ${({ uniformAspectRatio, setUniformAspectRatio }) =>
+    `${getMediaAspectRatio(uniformAspectRatio, setUniformAspectRatio)}`};
```

Comment:
nit: I would calculate the aspect ratio in the component and then pass it into the image so not have any calculations in the css itself.